autotest.pm

```diff
@@ -109,6 +109,58 @@ sub _debug_python_version () {
     bmwqemu::diag "Using python version " . $debug;
 }
 
+
+=head2 _lua_use
+
+    use("testapi");
+    use("some::perl::module", {"some_function", "another_function"});
+
+This function is the importer for loading Perl modules from Lua test modules.
+It imports everything that a Perl module exports into the global Lua scope.
+Optionally a table listing the desired functions can be passed.
+
+Imported hashes, scalars and arrays are copied as Inline:Lua doesn't support
+references for them.
+
+=cut
+
+sub _lua_use ($perlmod, $import = undef) {
+    my $inf = (defined $import) ? " qw(@$import)" : '';
+    eval("use ${perlmod}${inf}");
+    my @me = (defined $import) ? @{$import} : eval("\@{${perlmod}::EXPORT}");
+    foreach my $fn (@me) {
+        my $pref = '\&';    # import functions
+        if ($fn =~ m/^\$/) { $pref = '$'; $fn =~ s/\$//; }    # import scalars
+        elsif ($fn =~ m/^\%/) { $pref = '\%'; $fn =~ s/\%//; }    # import hashes
+        elsif ($fn =~ m/^\@/) { $pref = '\@'; $fn =~ s/\@//; }    # import arrays
+        lua_set($fn, eval("${pref}${perlmod}::${fn}"));
```

Comment:
It should be noted that imported variables won't work as expected.
If you import a variable in perl, then the imported variable is the same as the original one, meaning that if you assign a new value in one place, it's visible in the other place.

Here only the initial value will be set in the lua variables.

If we want to provide access to testapi variables, we should probably rather write wrapper functions, e.g. for `$realname`, `$user` and `$password`.

Reply:
The idea is to eventually get rid of those vars alltogether.
But at least the initial state is available to at least provide something.